apiary-metastore-listener/src/test/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListenerTest.java

```diff
@@ -106,10 +133,32 @@ public void onCreateTable() throws MetaException {
     snsListener.onCreateTable(event);
     verify(snsClient).publish(requestCaptor.capture());
     PublishRequest publishRequest = requestCaptor.getValue();
-    assertThat(publishRequest.getMessage(),
-        is("{\"protocolVersion\":\""
-            + PROTOCOL_VERSION
-            + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\"}"));
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"tableLocation\":\"s3://table_location\",\"tableParameters\":\"{MY_VAR_TWO=5, MY_VAR_ONE=true}\"}"));
+  }
+
+  @Test
+  public void tableParamRegexNotSet() throws MetaException {
+    environmentVariables.clear("TABLE_PARAM_FILTER");
+    ApiarySnsListener snsListener = new ApiarySnsListener(configuration, snsClient);
+
+    CreateTableEvent event = mock(CreateTableEvent.class);
+    when(event.getStatus()).thenReturn(true);
+    when(event.getTable()).thenReturn(table);
+
+    snsListener.onCreateTable(event);
+    verify(snsClient).publish(requestCaptor.capture());
+    PublishRequest publishRequest = requestCaptor.getValue();
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"tableLocation\":\"s3://table_location\",\"tableParameters\":\"{}\"}"));
+  }
+
+  @Test(expected = PatternSyntaxException.class)
+  public void tableParamRegexIsInvalid() throws MetaException {
+    environmentVariables.set("TABLE_PARAM_FILTER", "[");
+    new ApiarySnsListener(configuration, snsClient);
   }
 
```

Comment:
To be complete we would add another test of a table with no params at all - just to make sure that we get nothing back in this case (and not some unexpected NPE or some such).

Reply:
Added.